NetCDF classic only build fails with recent changes

My netcdf classic only build is now failing, due to recent changes:

```
/var/lib/jenkins/workspace/PIO_master_netcdf-4.4.1.1_classic_c_only/src/clib/pioc_support.c:1872: undefined reference to `nc_inq_unlimdims'
collect2: error: ld returned 1 exit status

```

--- a/doc/source/mach_walkthrough.txt
+++ b/doc/source/mach_walkthrough.txt
@@ -431,7 +431,7 @@ build from the command line, including tests:
 
 <pre>
 cd build
-rm -rf * && CFLAGS='-Wall -g' FFLAGS=-g CC=mpicc FC=mpifort cmake -DNetCDF_C_PATH=/usr/local/netcdf-4.4.1 -DNetCDF_Fortran_PATH=/usr/local/netcdf-fortran-4.4.4 -DPnetCDF_PATH=/usr/local/pnetcdf-1.8.1 -DPIO_ENABLE_LOGGING=On .. && make VERBOSE=1 all tests check
+rm -rf * && CFLAGS='-Wall -g' FFLAGS=-g CC=mpicc FC=mpifort cmake -DNetCDF_C_PATH=/usr/local/netcdf-c-4.5.0-rc1_mpich-3.2 -DNetCDF_Fortran_PATH=/usr/local/netcdf-fortran-4.4.4_mpich-3.2 -DPnetCDF_PATH=/usr/local/pnetcdf-1.8.1_mpich-3.2 -DPIO_HDF5_LOGGING=On -DPIO_USE_MALLOC=On .. && make VERBOSE=1 all tests check
 </pre>
 
 <p>Note the optional CFLAGS=-g which allows the use of a debugger

--- a/src/clib/pioc_support.c
+++ b/src/clib/pioc_support.c
@@ -1864,6 +1864,7 @@ int PIOc_createfile_int(int iosysid, int *ncidp, int *iotype, const char *filena
  */
 int check_unlim_use(int ncid)
 {
+#ifdef _NETCDF4
     int nunlimdims; /* Number of unlimited dims in file. */
     int nvars;       /* Number of vars in file. */
     int ierr;        /* Return code. */
@@ -1902,7 +1903,8 @@ int check_unlim_use(int ncid)
             }
         }
     }
-
+#endif /* _NETCDF4 */
+    
     return PIO_NOERR;
 }
 
@@ -2038,7 +2040,7 @@ int PIOc_openfile_retry(int iosysid, int *ncidp, int *iotype, const char *filena
                     break;                    
             }
             break;
-#endif
+#endif /* _NETCDF4 */
 
         case PIO_IOTYPE_NETCDF:
             if (ios->io_rank == 0)

